### PR TITLE
Utilities: Add -d option to gzip

### PR DESCRIPTION
**Pull request by itamar8910**
This option specifies that we should perform decompression instead of compression.

We use it in the build process to extract the PCI & USB databases.

cc @IdanHo

### CODE REVIEW

File: Userland/Utilities/gzip.cpp [hunk 1]
```diff
@@ -37,26 +48,32 @@ int main(int argc, char** argv)
...
-            warnln("Failed gzip compressing input file");
+        AK::Optional<ByteBuffer> output_bytes;
+        if (decompress) {
+            output_bytes = Compress::GzipDecompressor::decompress_all(file->bytes());
+        } else {
+            output_bytes = Compress::GzipCompressor::compress_all(file->bytes());
+        }
+
+        if (!output_bytes.has_value()) {
+            warnln("Failed gzip {} input file", decompress ? "decompressing"sv : "compressing"sv);
```
Comment (IdanHo) [inline Userland/Utilities/gzip.cpp:59]: Not sure i love this pattern, but :shrug:

**Review by IdanHo — APPROVED** [review]
LGTM

File: Userland/Utilities/gzip.cpp (hunk 1 above)
Comment (itamar8910) [reply Userland/Utilities/gzip.cpp:59]: yeah okay, you'd prefer plain `if/else`?

Comment (IdanHo) [reply Userland/Utilities/gzip.cpp:59]: Personally yes, but they're equivalent, so it's not a blocker to me

**Pull request merged**